osu.Game/Tests/Visual/Multiplayer/TestMultiplayerClient.cs

```diff
@@ -282,13 +296,78 @@ public override Task ChangeUserMods(IEnumerable<APIMod> newMods)
             return Task.CompletedTask;
         }
 
+        private CancellationTokenSource? countdownFinishSource;
+        private CancellationTokenSource? countdownStopSource;
+        private Task countdownTask = Task.CompletedTask;
+
+        public void FinishCountdown() => countdownFinishSource?.Cancel();
+
         public override async Task SendMatchRequest(MatchUserRequest request)
         {
             Debug.Assert(Room != null);
             Debug.Assert(LocalUser != null);
 
             switch (request)
             {
+                case StartMatchCountdownRequest matchCountdownRequest:
+                    countdownStopSource?.Cancel();
+
+                    var stopSource = countdownStopSource = new CancellationTokenSource();
+                    var finishSource = countdownFinishSource = new CancellationTokenSource();
+                    var cancellationSource = CancellationTokenSource.CreateLinkedTokenSource(stopSource.Token, finishSource.Token);
+                    var countdown = new MatchStartCountdown { TimeRemaining = matchCountdownRequest.Duration };
+
+                    Task lastCountdownTask = countdownTask;
+                    countdownTask = start();
+
+                    async Task start()
```

Comment:
Any reason for not using `countdownTask = Task.Run(async () => ...)`? I think it reads marginally better (at first my brain had a bit of trouble parsing what was going on here).

Reply:
Because then you need to unwrap.

Reply:
Hmm, okay, let's go with this for now.